### PR TITLE
add a singleton handle, modify storage proxy to apply updates and for…

### PR DESCRIPTION
**Pull request by galganif**
…ward them to the handle

### CODE REVIEW

File: src/runtime/storageNG/handle.ts [hunk 1]
```diff
@@ -80,4 +105,78 @@ export class CollectionHandle<T> extends Handle<CRDTCollectionTypeRecord<T>> {
   toList(): T[] {
     return [...this.storageProxy.getParticleView()];
   }
+
+  onUpdate(ops: CollectionOperation<T>[]) {
+    for (const op of ops) {
+      // Pass to change up to the particle.
```
Comment (shans) [inline src/runtime/storageNG/handle.ts]: Pass the change?

File: src/runtime/storageNG/storage-proxy.ts [hunk 2]
```diff
@@ -34,4 +38,32 @@ export class StorageProxy<T extends CRDTTypeRecord> {
...
+        for (const op of message.operations) {
+          this.crdt.applyOperation(op);
+        }
+        for (const handle of this.handles) {
+          if (handle.options.notifyUpdate) {
+            handle.onUpdate(message.operations);
+          }
+        }
+        break;
+      default:
```
Comment (shans) [inline src/runtime/storageNG/storage-proxy.ts]: there's RequestSync too, which should be handled by *sending* the local model.

File: src/runtime/storageNG/handle.ts (hunk 1 above)
Comment (galganif) [reply src/runtime/storageNG/handle.ts]: done

File: src/runtime/storageNG/storage-proxy.ts (hunk 2 above)
Comment (galganif) [reply src/runtime/storageNG/storage-proxy.ts]: added TODO

**Pull request merged**